src/net.rs

```diff
@@ -167,6 +189,12 @@ pub struct NonceRequest {
     pub sign_id: u64,
     /// Signing round iteration ID
     pub sign_iter_id: u64,
+    /// The message to sign
+    pub message: Vec<u8>,
+    /// Whether to make a taproot signature
+    pub is_taproot: bool,
```

Comment:
Why was it necessary to chain `is_taproot` and `merkle_root` though this entire message flow?

Reply:
Because we could be triggering a signing round without the info available to trigger it. A coordinator will do "start_signing_round" with these parameters passed in. This triggers a nonce request. An "observing signer" will not see the original "start_signing_round" but only sees the "nonce_request" which tells them "Oh I should trigger a signing round (i.e. I should duplicate this state) but they do not have the necessary is_taproot and merkle root. So I pass this info in the net message itself. Note that "observing" coordinators triggering signing rounds will send a packet nonce request but all other signers and the coordinator will ignore it as they are ot the "designated" coordinator. But it will still be a valid nonce request and will update the"obseving" coordinator's internal state correctly to match the real coordinator's internal state.